lib/handler/chunked.c

```diff
@@ -78,22 +78,22 @@ static void on_setup_ostream(h2o_filter_t *self, h2o_req_t *req, h2o_ostream_t *
     if (req->is_subrequest)
         goto Next;
 
-    /* do nothing if not HTTP/1.1 or content-length is known */
-    if (req->res.content_length != SIZE_MAX || req->version != 0x101)
+    /* do nothing if not HTTP/1.1 */
+    if (req->version != 0x101)
         goto Next;
+    /* do nothing if content-length is known */
+    if (req->res.content_length != SIZE_MAX)
+        goto NextWithoutTrailer;
     /* RFC 2616 4.4 states that the following status codes (and response to a HEAD method) should not include message body */
     if ((100 <= req->res.status && req->res.status <= 199) || req->res.status == 204 || req->res.status == 304)
-        goto Next;
+        goto NextWithoutTrailer;
     else if (h2o_memis(req->input.method.base, req->input.method.len, H2O_STRLIT("HEAD")))
-        goto Next;
+        goto NextWithoutTrailer;
+
     /* we cannot handle certain responses (like 101 switching protocols) */
     if (req->res.status != 200) {
         req->http1_is_persistent = 0;
-        goto Next;
     }
-    /* skip if content-encoding header is being set */
-    if (h2o_find_header(&req->res.headers, H2O_TOKEN_TRANSFER_ENCODING, -1) != -1)
-        goto Next;
```

Comment:
Please do not remove this check.

The check exists for the case when a CGI returns it's own `transfer-encoding`. That is not disallowed, and in such case, we cannot append a trailer, since it is either the CGI that emits the trailer, or it is an unknown transfer encoding.

Reply:
In our previous discussion, we found that there're no chance that transfer-encoding header is filled (because fastcgi and other handlers drop it by seeing `proxy_should_drop_for_res` flag), so this is redundant check. Am I right?

Reply:
Thank you for the clarification.